contrib/win32/win32compat/misc.c

```diff
@@ -2119,3 +2124,97 @@ strrstr(const char *inStr, const char *pattern)
 
 	return last;
 }
+
+int
+add_mark_of_web(const char* filename)
+{
+	if (motw_zone_id < 5) {
+		char* fileStreamPath = NULL;
+		size_t fileStreamPathLen = strlen(filename) + strlen(":Zone.Identifier") + 1;
+
+		fileStreamPath = malloc(fileStreamPathLen * sizeof(char));
+
+		if (fileStreamPath == NULL) {
+			return -1;
+		}
+
+		sprintf_s(fileStreamPath, fileStreamPathLen, "%s:Zone.Identifier", filename);
+
+		char* zoneIdentifierStr = NULL;
+		size_t zoneIndentifierLen = strlen("[ZoneTransfer]\nZoneId=") + 1 + 1;
+
+		zoneIdentifierStr = malloc(zoneIndentifierLen * sizeof(char));
+
+		if (zoneIdentifierStr == NULL) {
+			return -1;
```

Comment:
`fileStreamPath` won't be freed in this case.  I think we should set status here and go to `cleanup`.  Or we could just free `fileStreamPath` here.